src/react-apps/applications/shared/src/navigation/NavMenu.tsx

```diff
@@ -0,0 +1,257 @@
+import Collapse from '@material-ui/core/Collapse';
+import Divider from '@material-ui/core/Divider';
+import Drawer from '@material-ui/core/Drawer';
+import List from '@material-ui/core/List';
+import ListItem from '@material-ui/core/ListItem';
+import ListItemIcon from '@material-ui/core/ListItemIcon';
+import ListItemText from '@material-ui/core/ListItemText';
+import { createStyles, withStyles, WithStyles, Theme } from '@material-ui/core/styles';
+import classNames from 'classnames';
+import * as React from 'react';
+import { leftNavMenuSettings } from './leftNavMenuSettings';
+
+const drawerWidth = 250;
+
+const styles = (theme: Theme) => createStyles({
+  root: {
+    display: 'flex'
+  },
+  menuButton: {
+    marginLeft: 12,
+    marginRight: 36,
+  },
+  hide: {
+    display: 'none'
+  },
+  drawer: {
+    width: drawerWidth,
+    flexShrink: 0,
+    whiteSpace: 'nowrap',
+    top: '64 !important',
+  },
+  drawerOpen: {
+    width: drawerWidth,
+    transition: theme.transitions.create('width', {
+      easing: theme.transitions.easing.sharp,
+      duration: theme.transitions.duration.enteringScreen,
+    })
+  },
+  drawerClose: {
+    transition: theme.transitions.create('width', {
+      easing: theme.transitions.easing.sharp,
+      duration: theme.transitions.duration.leavingScreen,
+    }),
+    overflowX: 'hidden',
+    width: theme.spacing.unit * 7 + 1,
+    [theme.breakpoints.up('sm')]: {
+      width: theme.spacing.unit * 9 + 1
+    }
+  },
+  paper: {
+    top: 69,
+    position: 'absolute',
+    background: 'lightgrey',
+  },
+  toggleButton: {
+    position: 'fixed',
+    bottom: 0,
+    left: 0,
+  },
+  toggleButtonClosed: {
+    width: theme.spacing.unit * 7 + 1,
+    [theme.breakpoints.up('sm')]: {
+      width: theme.spacing.unit * 9 + 1,
+    }
+  },
+  toggleButtonOpen: {
+    width: drawerWidth,
+  },
+  nested: {
+    paddingLeft: theme.spacing.unit * 4,
+    fontSize: '20px',
+  },
+  menuItemText: {
+    fontSize: '1.5rem',
+  },
+  menuSubItemText: {
+    fontSize: '1.2rem',
+  },
+  toggleMenuText: {
+    color: 'white',
+  },
+  toggleMenu: {
+    background: 'grey',
+  },
+  menuItemTextClosed: {
+    visibility: 'hidden',
+  },
+  selectedMenuItem: {
+    color: 'white',
+    textDecoration: 'underline',
+    background: 'grey',
+  },
+  selectedMenuItemText: {
+    color: 'white',
+    fontWeight: 'bold',
+  },
+});
+
+export interface INavMenuState {
+  open: boolean;
+  openSubMenus: number[];
+}
+
+export interface INavMenuProps extends WithStyles<typeof styles> { }
+
+class NavMenu extends React.Component<any, any> {
+  public state = {
+    open: false,
+    openSubMenus: [] as number[],
+    selectedMenuItem: '',
+  }
+
+  public handleNavMenuOpen = () => {
+    this.setState({ open: true });
+  };
+
+  public handleNavMenuClose = () => {
+    this.setState({ open: false });
+  };
+
+  public toggleNavMenu = () => {
+    if (this.state.open) {
```

Comment:
`this.setState({open: !this.state.open});` 
instead of if and the two other functions

Reply:
had an old version of this code for some reason.. Updated to the latest version from master.